POM.xml and other amendments

paulswithers opened the pull request.

rPraml commented on domino/core/pom.xml:
```diff
@@ -20,6 +20,12 @@
 				<extensions>true</extensions>
 				<configuration>
 					<singleton>true</singleton>
+					<manifestLocation>META-INF</manifestLocation>
```
As far as I understand this setting, the MANIFEST.MF is put now into the /META-INF directory and no longer in the /target/.... directory

paulswithers replied:
That seems to be the case, but they're still created in the updateSite's jars. Is that because those jars are created differently? Does the location outside of /target/ cause a problem?

rPraml replied:
don't know. The /target will be deleted on "mvn clean", so this manifest is never deleted. As it is in .gitignore, it should be no problem

rPraml commented on domino/core/pom.xml:
```diff
@@ -20,6 +20,12 @@
 				<configuration>
 					<singleton>true</singleton>
+					<manifestLocation>META-INF</manifestLocation>
+					<source>${compiler}</source>
```
Do we need compiler settings here?

rPraml commented on domino/core/pom.xml:
```diff
@@ -34,12 +40,11 @@
   						<!-- BND generated import-packages does not work in DominoServer -->
-  						<!-- 
+  						
   							so we import just lotus.domino	(do not know why, it was several times added to original MANIFEST.MF 
```
This line should be removed or commented out

rPraml commented on domino/core/pom.xml:
```diff
@@ -51,8 +56,7 @@
 							org.openntf.formula;bundle-version="1.5.0";visibility=reexport
-						</Require-Bundle>
--->
+ 						</Require-Bundle>
```
I'm not happy about using "Require-Bundle" because it is deprecated. But I think there is no way around, because the Domino OSGI-Implementation is very old

rPraml commented on domino/core/src/main/java/org/openntf/domino/design/XspXmlContent.java:
```diff
@@ -19,7 +19,6 @@
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.lang.reflect.InvocationTargetException;
-import java.nio.charset.StandardCharsets;
```
Made the same fix.

rPraml commented on domino/core/src/main/java/org/openntf/domino/design/impl/StyleSheet.java:
```diff
@@ -18,7 +18,6 @@
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
```
Made the same fix

rPraml commented on domino/formula/pom.xml:
```diff
@@ -31,6 +31,12 @@
 					<singleton>true</singleton>
+					<manifestLocation>META-INF</manifestLocation>
+					<source>${compiler}</source>
+					<target>${compiler}</target>
```
Maybe we can move the $compiler configuration up to the domino/root and put the default configuration of the maven-bundle-plugin  in a <pluginManagement> section

rPraml commented on domino/graph/pom.xml:
```diff
@@ -5,8 +5,15 @@
 		<artifactId>root</artifactId>
 		<version>1.5.1-SNAPSHOT</version>
 	</parent>
+	<groupId>org.openntf.domino</groupId>
```
this line is not needed. should be inherited from parent

rPraml commented on domino/graph/pom.xml:
```diff
@@ -5,8 +5,15 @@
+	<groupId>org.openntf.domino</groupId>
 	<artifactId>graph</artifactId>
+	
+	<version>1.5.1-SNAPSHOT</version>
```
Version should also be inherited from parent

rPraml commented on domino/xsp/domino-faces-config.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
These files should be synchronized with the ones in src/main/resources and ignored with .gitignore

paulswithers replied:
Looks like this could be the answer http://stackoverflow.com/questions/8386543/copying-file-from-one-project-to-another-in-maven. I'll give it a go.

paulswithers commented:
Completed those changes, copying files

paulswithers committed:
Was not recreating MANIFEST>MF correctly

paulswithers committed:
Fixes for javolution-core-java poms - amendment to pom.xml for core to point to correct bundle and addition to javolution-core-java pom.xml to not lazy-load. Configuration class, which is triggered on load, uses it anyway

paulswithers committed:
Amendments to pom.xml so frames and graph are available as plugins
HOWEVER, lots of constraints unresolved - missing imported packages, but relevant classes don't appear to be available as plugins - not sure on way forward

paulswithers committed:
gitignore and plugin.xml, missed from last commit

paulswithers committed:
Merge remote-tracking branch 'origin/jesse' into paul_new

paulswithers committed:
Fix for OpenLog logging correct database location

paulswithers committed:
Fix for ViewColumn.getIndex with calendar views

paulswithers committed:
ODAPlatform was not setting isStarted_
